sound/soc/sof/intel/hda.c

```diff
@@ -1644,11 +1644,6 @@ struct snd_soc_acpi_mach *hda_machine_select(struct snd_sof_dev *sdev)
 			add_extension = true;
 		}
 
-		if (mach->link_mask) {
-			mach->mach_params.links = mach->links;
-			mach->mach_params.link_mask = mach->link_mask;
-		}
-
```

Comment:
what makes you think this is wrong? This doesn't look like a bug but a feature to me...

Keep in mind that we initially had platforms where we detected the machine driver using an _HID related to the presence of an SSP, but some SoundWire links were also used.

IIRC this was with the RT1308 based on I2S and RT711 based on SoundWire. That code is still present in the machine driver.

The SoundWire machine detection in  hda_sdw_machine_select() only works if there is NO I2S support.

Reply:
@aiChaoSONG that's a good learning I guess, nothing wrong with asking for reviews.

so the code progresses by steps
a) SSP,  SoundWire, DMIC may be supported
b) SoundWire, DMIC may be supported
c) HDaudio, DMIC is supported.

We don't support other setups such as HDAudio+SoundWire or HDAudio+SSP because they don't really exist.

HDMI is not viewed as a specific interface, it's present in all 3 cases.